### ISSUE TITLE
Cannot create assistant with tool functions

### ISSUE THREAD

**Issue by k-polaschek** [skyscrapr/terraform-provider-openai#83]
There seems to be a problem with the creation of an assistant that is using tool functions.

``` tf
resource "openai_assistant" "bot" {
  name         = "Bot"
  model        = "gpt-3.5-turbo-0125"
  instructions = "You are the personal assistant for users who are using our app."
  tools = [
    {
      type = "function"
      function = {
        name = "get_additional_info"
        description = "Get additional information."
        parameters = jsonencode({
          type = "object",
          properties = {}
        })
      }
    }
  ]
}

```

Applying the terraform script raises the following error:

```
openai_assistant.test: Creating...
╷
│ Error: OpenAI Client Error
│ 
│   with openai_assistant.test,
│   on assistant.tf line 1, in resource "openai_assistant" "test":
│    1: resource "openai_assistant" "test" {
│ 
│ Unable to create assistant, got error: error, status code: 400, message: Missing required parameter: 'tools[0].function'.
```

Terraform plan works without any issue.

**Comment by skyscrapr** [comment]
Thanks for raising the issue. Will take a look later today. Looks like an easy fix.

**Comment by skyscrapr** [comment]
I've just released v1.3.0 which contains a fix for this issue. Support for functions and parameters in raw json should now be supported. I've added a test to verify a basic function.
